transaction_controller.go

```diff
@@ -0,0 +1,69 @@
+package amqp
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/go-amqp/internal/encoding"
+)
+
+// TransactionControllerOptions contains the optional settings for configuring a [TransactionController].
+type TransactionControllerOptions struct {
+	// Capabilities is the list of extension capabilities the sender supports.
+	Capabilities []string
```

Comment:
I'm thinking we should remove this in favor of a set of constants for configuring the requested txn-capability as it appears to be constrained to a small set of values. Or are there other capabilities we need to be able to specify?

Reply:
Interesting. Something like this?

```go
type TransactionControllerOptions struct {
   LocalTransactions bool
   DistributedTransactions bool
   PromotableTransactions bool
   MultipleTransactionsPerSession bool
   MultiSessionsPerTransaction bool
}
```

I like this idea, although I'm not sure if they are mutually exclusive so we'd probably not do any validation (just translate bools -> proper transaction flag).

Reply:
Per [1.3.5](http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-types-v1.0-os.html#doc-idp125312) I believe that choices are mutually exclusive.

Reply:
I think there's two bits we have to look at. The flags are mutually exclusive when picking a single value, but the Capabilities is specific setup to allow multiple:

http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transactions-v1.0-os.html#type-txn-capability:~:text=%3Cfield%20name%3D%22capabilities%22%20type%3D%22symbol%22%20requires%3D%22txn%2Dcapability%22%20multiple%3D%22true%22/%3E

So I think we can eliminate the array, in favor of the struct, but they are not inherently mutually exclusive.

Reply:
Ah yes I see now.
```
multiple="true"
```